scipy/optimize/_zeros_py.py

```diff
@@ -1393,3 +1394,222 @@ def toms748(f, a, b, args=(), k=1,
                           maxiter=maxiter, disp=disp)
     x, function_calls, iterations, flag = result
     return _results_select(full_output, (x, function_calls, iterations, flag))
+
+
+def chandrupatla(f, x0, x1, args=(), xtol=_xtol,
+                 rtol=_rtol, maxiter=_iter, callback=None):
+
+    """
+    As written in [Sachs2015]_ which in turn is based on Chandrupatla's
+    algorithm as described in [Scherer2018]_.
+    This allows vector arguments for x0, x1, and args
+
+    Parameters
+    ----------
+
+    f : callable
+        The function whose zero is wanted. It must be a function of a
+        single variable of the form ``f(x,a,b,c...)``, where ``a,b,c...``
+        are extra arguments that can be passed in the `args` parameter.
+    x0 : float, sequence, or ndarray
+        An initial estimate of the zero that should be somewhere near the
+        actual zero. If not scalar, then `func` must be vectorized and return
+        a sequence or array of the same shape as its first argument.
+    x1 : float, optional
+        Another estimate of the zero that should be somewhere near the
+        actual zero.
+    xtol : float, optional
+        Tolerance (absolute) for termination.
+    rtol : float, optional
+        Tolerance (relative) for termination.
+    maxiter : int, optional
+        If convergence is not achieved in `maxiter` iterations, an error is
+        raised. Must be >= 0.
+    callback : callable, optional
+        An optional user-supplied function, called after each iteration.
+        Called as ``callback(xk)``, where ``xk`` is the current value of `x0`.
+
+    Returns
+    -------
+    result : OptimizeResult
+        The optimization result represented as an ``OptimizeResult`` object.
+        Important attributes are: ``x`` the solution array, ``success`` an 
+        array of Boolean flags indicating if the optimizer exited successfully,
+        ``fun``, the current function value, ``nfev``, the number of function 
+        evaluations, and ``nit``, the number of iterations. See 
+        `OptimizeResult` for a description of other attributes.
```

Comment:
Let's include the attributes in a list like https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.milp.html does.

Something else that I think would be useful to include is the final bracket and lower/upper function values, perhaps as separate arrays `a`, `b`, `fa`, `fb`. It's verbose, but the user doesn't need to access them if they're not interested, and it's easy to prevent them from pretty-printing if we want.